diff --git a/src/frame.c b/src/frame.c
index 4eb42ac..1d1dd41 100644
--- a/src/frame.c
+++ b/src/frame.c
@@ -440,6 +440,10 @@ uvc_error_t uvc_uyvy2bgr(uvc_frame_t *in, uvc_frame_t *out) {
  */
 uvc_error_t uvc_any2rgb(uvc_frame_t *in, uvc_frame_t *out) {
   switch (in->frame_format) {
+#ifdef LIBUVC_HAS_JPEG
+    case UVC_FRAME_FORMAT_MJPEG:
+      return uvc_mjpeg2rgb(in, out);
+#endif
     case UVC_FRAME_FORMAT_YUYV:
       return uvc_yuyv2rgb(in, out);
     case UVC_FRAME_FORMAT_UYVY:
